### PR TITLE
tests: Add integration test for vfio-user with SPDK NVMe

### PR DESCRIPTION
**Pull request by likebreath**
This patch adds the integration for vfio-user with SPDK NVMe. 
It also enables `CONFIG_NVME_MULTIPATH` for the guest kernel
of x86_64 to resolve the observed guest hanging issue #3352.

Fixes: #3098

Signed-off-by: Bo Chen chen.bo@intel.com

### CODE REVIEW

**Comment by likebreath** [comment]
It looks like the vfio-user test is working fine for x86_64. 

The hanging issue on AArch64 is a separate and known issue, and I will take a look at it later.

File: tests/integration.rs
Comment (optimistyzy) [inline tests/integration.rs:5939]: Should be SPDK

File: tests/integration.rs
Comment (optimistyzy) [inline tests/integration.rs:5941]: Do we want to trace SPDK with "-e" "0xFFFF"

File: vfio_user/src/lib.rs
Comment (sboeuf) [inline vfio_user/src/lib.rs:301]: I'll let @rbradford decide here since he introduced the code with the `info` log level, and I don't want to go back and forth regarding this topic.

File: resources/linux-config-x86_64
Comment (sboeuf) [inline resources/linux-config-x86_64:1230]: Great, but could you elaborate a little bit in the commit message why we need this?

File: resources/linux-config-x86_64
Comment (sboeuf) [inline resources/linux-config-x86_64:1322]: Same here, elaborate a bit in your commit message by saying that these errors have been the cause for the guest to hang, especially when being used with an NVMe backend.

**Review by sboeuf — COMMENTED** [review]
Looks good, just a few nits, and also waiting for @rbradford approval before we can merge it.

File: tests/integration.rs
Comment (optimistyzy) [reply tests/integration.rs:5941]: This -e is used for tracing purpose. I think that you do not need that for test.  You can just delete -e 0xFFFF.

Meaning:  
 -e, --tpoint-group-mask <mask>
                           tracepoint group mask for spdk trace buffers (default 0x0, bdev 0x8, nvmf_tcp 0x20, blobfs 0x80, thread 0x400, nvme_pcie 0x800, all 0xffff)

I do not think that you need to use this for testing purpose.

File: vfio_user/src/lib.rs
Comment (rbradford) [reply vfio_user/src/lib.rs:301]: There is docs/logging.md that tries to give rationale for different levels of logging. But since this PR and also #3382 wants to downgrade it to `debug!()` then I guess it is too spammy for `info!()` so you can go ahead and downgrade it.

File: resources/linux-config-x86_64
Comment (likebreath) [reply resources/linux-config-x86_64:1230]: Of course. Commit message updated.

File: resources/linux-config-x86_64
Comment (likebreath) [reply resources/linux-config-x86_64:1322]: Ditto

File: tests/integration.rs
Comment (rbradford) [inline tests/integration.rs:5929]: Can you put use `guest.tmp_dir` ?

Comment (likebreath) [reply tests/integration.rs]: PR updated.

**Review by sboeuf — APPROVED** [review]

File: tests/integration.rs
Comment (sboeuf) [inline tests/integration.rs:5924]: Typo: you have two `Signed-off-by: Bo Chen <chen.bo@intel.com>` in your commit `tests: Add integration test for vfio-user with SPDK NVMe`

Comment (likebreath) [reply tests/integration.rs:5924]: Good catch. Fixed!

**Review by MrXinWang — APPROVED** [review]
We will take a look on the AArch64 platform, thanks for the change and effort @likebreath :))

**Review by rbradford — APPROVED** [review]

**Pull request merged**
